### PR TITLE
upload resource - concept added

### PR DESCRIPTION
**Pull request by sharad-sharma-msft**
Assignment service only exposes API for setting up SPO resources folder. However uploading the assignment or submission resource is done using direct API graph calls. This flow needs to be documented to allow graph callers to effectively exercise the flow end to end.

### CODE REVIEW

File: concepts/education-upload-resource-overview.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+description: "Upload a file to a given educationAssignment or educationSubmission resource."
+localization_priority: Normal
+author: "sharad-sharma-msft"
+ms.prod: "education"
+doc_type: apiPageType
+---
+
+# Education upload resource overview
+
+Resources are integral part of [educationAssignment](/graph/api/resources/educationassignment.md) and [educationSubmission](/graph/api/resources/educationsubmission.md).
```
Comment (ssriniva-msft) [inline concepts/education-upload-resource-overview.md:12]: are **a** integral....

File: concepts/education-upload-resource-overview.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+| Action  |
+|:--------|
+| Setup a SharePoint folder for a given [educationAssignment](/graph/api/resources/educationAssignment.md) resource|
+| Setup a SharePoint folder for a given [educationSubmission](/graph/api/resources/educationSubmission.md) resource|
+
+## Upload a resource
+
+At this point, we are assuming that you have setup the relevant resource folder already. 
+This returns a model with one of the property called `resourcesFolderUrl`.
```
Comment (ssriniva-msft) [inline concepts/education-upload-resource-overview.md:28]: The setup process

File: concepts/education-upload-resource-overview.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+```http
+{
+    ...
+
+    "resourcesFolderUrl": "https://graph.microsoft.com/v1.0/drives/b!6SQl0y4WHkS2P5MeIsSGpKwfynEIaD1OvPVeH4wbOp_1uyhNwJMSSpseJneB7Z4F/items/01YT2AIJRQLVYT24IWWFAJHMRRNYCB3GFA"
+    ...
+}
+```
+
+Following steps will allow you to upload a resource/file to the relevant resource folder. For simplicity, we are going to upload an SPO resource to the assignment.
```
Comment (ssriniva-msft) [inline concepts/education-upload-resource-overview.md:38]: What other options are there other than "SPO resource"? Isn't this the only option?

File: concepts/education-upload-resource-overview.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+### Step 1 - Construct the upload url
+Build the url for uploading content in a specific format. The format is `{resourcesFolderUrl}:/{Name of new file}:/content`, 
+e.g. based on sample `resourcesFolderUrl`, the upload url will be:
```
Comment (ssriniva-msft) [inline concepts/education-upload-resource-overview.md:42]: base on **the** sample

File: concepts/education-upload-resource-overview.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+#### Example Response
+```http
+{
+    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#drives('b%216SQl0y4WHkS2P5MeIsSGpKwfynEIaD1OvPVeH4wbOp_1uyhNwJMSSpseJneB7Z4F')/items/$entity",
+    "@microsoft.graph.downloadUrl": "...",
+    "createdDateTime": "2021-03-11T18:49:47Z",
+    "eTag": "\"{EDD00CE7-B74C-4C3E-BA3E-484CB41EF31D},1\"",
+    "id": "01YT2AIJU7DAXTU6XLOJGYWYMTGM5JT5UQ",
+    "lastModifiedDateTime": "2021-03-11T18:49:47Z",
+    "name": "NewFile.png",
```
Comment (ssriniva-msft) [inline concepts/education-upload-resource-overview.md:67]: This name doesn't match the input name in the request URL. Is that expected?

File: concepts/education-upload-resource-overview.md (hunk 3 above)
Comment (sharad-sharma-msft) [reply concepts/education-upload-resource-overview.md:38]: emphasis on assignment (we could show a sample for assignment resource upload or submission resource upload)

File: concepts/education-upload-resource-overview.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    "webUrl": "https://contososdorg.sharepoint.com/sites/GraphTest/Class%20Files/Assignments/Test%20File%20Distribution/NewFile.png",
```
Comment (ssriniva-msft) [inline concepts/education-upload-resource-overview.md:68]: Same name comment as above

File: concepts/education-upload-resource-overview.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    "cTag": "\"c:{EDD00CE7-B74C-4C3E-BA3E-484CB41EF31D},2\"",
+    "size": 2302233,
+    "createdBy": {
+        "application": {
+            "id": "de8bc8b5-d9f9-48b1-a8ad-b748da725064",
+            "displayName": "Graph explorer"
```
Comment (ssriniva-msft) [inline concepts/education-upload-resource-overview.md:74]: Is this accurate?

File: concepts/education-upload-resource-overview.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+```
+
+#### Example Response
+```http
+{
+    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#education/classes('b07edbef-7420-4b3d-8f7c-d599cf21e069')/assignments('48b80dff-452a-4108-bd85-fa0d84e39d0a')/resources/$entity",
+    "assignmentResourceUrl": null,
+    "id": "ff1aafe4-ae89-49c3-8366-4b509f640d6a",
+    "resource": {
+        "@odata.type": "#microsoft.graph.educationWordResource",
```
Comment (ssriniva-msft) [inline concepts/education-upload-resource-overview.md:149]: The resource in the previous step was a .png file - this says it is a Word resource

File: concepts/education-upload-resource-overview.md (hunk 7 above)
Comment (sharad-sharma-msft) [reply concepts/education-upload-resource-overview.md]: omitted

**Comment by JarbasHorst** [comment]
Hi @sharad-sharma-msft 

I believe your PR is missing the toc entry for the new concept topic. Can you please add the toc entry to the concept TOC under Learn/Education?

image

Thanks,
Jarbas

**Review by JarbasHorst — CHANGES_REQUESTED** [review]

File: concepts/education-upload-resource-overview.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
+---
+title: "Education upload resource overview"
+description: "Upload a file to a given educationAssignment or educationSubmission resource."
```
[the remark this answers is not in the document]
Comment (sharad-sharma-msft) [reply concepts/education-upload-resource-overview.md]: Hi @JarbasHorst , I have addressed your comments. Please review and help close the comments.

**Comment by JarbasHorst** [comment]
Hello @sharad-sharma-msft,

thanks for applying part of my feedback. Can you please review the comments that are still open?

Thanks,
Jarbas

**Comment by JarbasHorst** [comment]
Hi @sharad-sharma-msft,

why did you [fix the links](https://github.com/microsoftgraph/microsoft-graph-docs/pull/13359/commits/9890dd52e2c5e6fc57cac2f345b41a6748a3e6ae)? The first heading within an article should be an H2, not an H1. Please revert your changes.

Thanks,
Jarbas

File: concepts/education-upload-resource-overview.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+
+### Step 2 - Upload the resource to SPO
+
+PUT `{upload Url}` to upload the content. 
+
+Refer to [this](/graph/api/driveitem-createuploadsession?view=graph-rest-1.0&preserve-view=true) documentation for more details.
+
+#### Example Request
+```http
+PUT https://graph.microsoft.com/v1.0/drives/b!6SQl0y4WHkS2P5MeIsSGpKwfynEIaD1OvPVeH4wbOp_1uyhNwJMSSpseJneB7Z4F/items/01YT2AIJRQLVYT24IWWFAJHMRRNYCB3GE2:/MyPictureFile.png:/content
```
[the remark this answers is not in the document]
Comment (sharad-sharma-msft) [reply concepts/education-upload-resource-overview.md:56]: good idea!

File: concepts/education-upload-resource-overview.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+`{drive-id}` - drive ID from the request URL used in step 2 = b!6SQl0y4WHkS2P5MeIsSGpKwfynEIaD1OvPVeH4wbOp_1uyhNwJMSSpseJneB7Z4F
+`{item-id}` - item ID from the response body obtained in step 2 = 01YT2AIJU7DAXTU6XLOJGYWYMTGM5JT5UQ
+Here is what a **fileUrl** built based on the format described looks like:
+https://graph.microsoft.com/v1.0/drives/b!6SQl0y4WHkS2P5MeIsSGpKwfynEIaD1OvPVeH4wbOp_1uyhNwJMSSpseJneB7Z4F/items/01YT2AIJU7DAXTU6XLOJGYWYMTGM5JT5UQ
+
+### Step 4. Create educationAssignmentResource
+This step describes how to upload a SharePoint resource to an assignment resources folder.
+
+Use the `fileUrl` constructed in the previous step in the request body to [POST Assignment Resources](/graph/api/educationassignment-post-resources)
```
Comment (JarbasHorst) [inline concepts/education-upload-resource-overview.md:127]: For consistency, replace "POST Assignment Resources" with "Create educationAssignmentResource"

File: concepts/education-upload-resource-overview.md [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+    "resource": {
+        "@odata.type": "#microsoft.graph.educationFileResource",
+        "fileUrl": "https://graph.microsoft.com/v1.0/drives/b!6SQl0y4WHkS2P5MeIsSGpKwfynEIaD1OvPVeH4wbOp_1uyhNwJMSSpseJneB7Z4F/items/01YT2AIJU7DAXTU6XLOJGYWYMTGM5JT5UQ",
+        "displayName": "Parts of a Sonnet"
+    }
+}
+```
+
+#### Example response
+```http
```
Comment (JarbasHorst) [inline concepts/education-upload-resource-overview.md:144]: For consistency with your previous examples, please add the following HTTP headers to this example response:

```http
HTTP/1.1 201 Created
Content-type: application/json

{
...
}
```

File: concepts/education-upload-resource-overview.md (hunk 11 above)
Comment (JarbasHorst) [inline concepts/education-upload-resource-overview.md:127]: Please, add a period at the end of the sentence.

**Review by JarbasHorst — CHANGES_REQUESTED** [review]

**Comment by cristobal-buenrostro** [comment]
pretty small differences in Response at Step#4, specifically these properties.
```
"assignmentResourceUrl": null,
"distributeForStudentWork": false,
```
response from Graph explorer
```
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#education/classes('f4a941ff-9da6-4707-ba5b-0eae93cad0b4')/assignments('9018ae7a-9953-4796-a152-4c54e0910922')/resources/$entity",
    "distributeForStudentWork": false,
    "id": "eec7f642-9d9a-406f-bbae-4b3b2c12e273",
    "resource": {
        "@odata.type": "#microsoft.graph.educationFileResource",
        "displayName": "First file uploaded as Education resource by t-cristobalb",
        "createdDateTime": "2021-07-16T23:41:53.9378423Z",
        "lastModifiedDateTime": "2021-07-16T23:41:53.9378423Z",
        "fileUrl": "https://graph.microsoft.com/v1.0/drives/b!DPA6q59Tw0mtgmyXRUmrQRqBZTesG-lMkl1cBmvvMeU6BLWBcGc_R6UgCKyYyTin/items/016XPCQEA5VVDIMU4BSFG3VBI37MPHZ3OE",
        "createdBy": {
            "application": null,
            "device": null,
            "user": {
                "id": "f3a5344e-dbde-48b0-be24-b5b62a243836",
                "displayName": null
            }
        },
        "lastModifiedBy": {
            "application": null,
            "device": null,
            "user": {
                "id": "f3a5344e-dbde-48b0-be24-b5b62a243836",
                "displayName": null
            }
        }
    }
}
```

**Review by Lauragra — APPROVED** [review]

**Pull request merged**